### PR TITLE
[DOC release] document {{input}} helper

### PR DESCRIPTION
**Pull request by EricSchank**
Document the `{{input}}` helper.

### CODE REVIEW

**Comment by rwjblue** [comment]
I could have sworn we had docs for `{{input}}` at one point.  They could have been lost in the shuffle on the 1.13 rendering engine rewrite.

**Comment by rwjblue** [comment]
Ahh, yes here they are: https://github.com/emberjs/ember.js/blob/v1.12.1/packages/ember-htmlbars/lib/helpers/input.js#L14-L188

Can you review the above, and incorporate what makes sense?

**Comment by EricSchank** [comment]
I've incorporated a bunch of the content from the 1.12 input helper.

**Comment by EricSchank** [comment]
@rwjblue 
Any other edits you'd suggest on this one?
Thanks!

File: packages/ember-htmlbars/lib/keywords/input.js [hunk 1]
```diff
@@ -1,6 +1,157 @@
...
+
+  ```javascript
+  Ember.TextField.reopen({
+    attributeBindings: ['data-error']
+  });
+  ```
+
+  Keep in mind when writing `Ember.TextField` subclasses that `Ember.TextField`
+  itself extends `Ember.Component`, meaning that it does NOT inherit
+  the `controller` of the parent view.
```
Comment (mixonic) [inline packages/ember-htmlbars/lib/keywords/input.js]: Please replace this paragraph with:

```
Keep in mind when writing `Ember.TextField` subclasses that `Ember.TextField`
itself extends `Ember.Component`. Expect isolated component semantics, not
legacy 1.x view semantics (like `controller` being present).
```

**Comment by mixonic** [comment]
@EricSchank if you can make those changes I would love to click merge!

File: packages/ember-htmlbars/lib/keywords/input.js [hunk 2]
```diff
@@ -1,6 +1,157 @@
...
+  {{input value="987"}}
+  ```
+
+  renders as:
+
+  ```HTML
+  <input type="text" value="987" />
+  ```
+
+  ## Text field
```
Comment (mixonic) [inline packages/ember-htmlbars/lib/keywords/input.js]: Please make this `###`

File: packages/ember-htmlbars/lib/keywords/input.js [hunk 3]
```diff
@@ -1,6 +1,157 @@
...
+  ```
+
+  Keep in mind when writing `Ember.TextField` subclasses that `Ember.TextField`
+  itself extends `Ember.Component`, meaning that it does NOT inherit
+  the `controller` of the parent view.
+
+  See more about [Ember components](/api/classes/Ember.Component.html)
+
+
+  ## Checkbox
```
Comment (mixonic) [inline packages/ember-htmlbars/lib/keywords/input.js]: Please make this `###`

**Comment by EricSchank** [comment]
@mixonic Thanks for the feedback, I've included all your changes in the latest commit.

**Comment by mixonic** [comment]
@EricSchank thank you sir :-)

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**